File: code/e07.py
from math import sqrt
x = float(raw_input("Enter a number x and I will compute sqrt(x):"))
niter = int(raw_input("How many iterations? "))
i = 0
xi = 10
while (i < niter):
    xi = xi - ((xi*xi)-x) / (2*xi)
    i = i + 1
print "after %d iterations, sqrt(%.5f) = %.5f" % (i, x, xi)
err = xi - sqrt(x)
print "the error is " + str(err)

File: code/e07.c
// compile with: gcc -o e07 e07.c

#include <stdio.h>
#include <math.h>

int main(int argc, char *argv[]) {

  printf("Enter a number x and I will compute sqrt(x): ");
  double x;
  scanf("%lf", &x);
  printf("How many iterations? ");
  int niter;
  scanf("%d", &niter);
  int i = 0;
  double xi = 10;
  while (i < niter) {
    xi = xi - ((xi*xi)-x) / (2*xi);
    i = i + 1;
  }
  printf("after %d iterations, sqrt(%.5f) = %.5f\n", i, x, xi);
  double err = xi - sqrt(x);
  printf("the error is %.20f\n", err);

  return 0;
}
